### PR TITLE
Fix: datetime parameters were not using a date picker.

### PR DESCRIPTION
**Pull request by arikfr**

### CODE REVIEW

File: client/app/services/query.js [hunk 1]
```diff
@@ -43,6 +43,40 @@ class QueryResultError {
...
+    }
+
+    return this.value;
+  }
+
+  set ngModel(value) {
+    if (this.type === 'date') {
+      this.value = moment(value).format('YYYY-MM-DD');
+      this.$$value = moment(this.value).toDate();
+    } else if (this.type === 'datetime-local' || this.type === 'datetime-with-seconds') {
```
Comment (rockwotj) [inline client/app/services/query.js]: Why is there two versions of this if they are handled the same? If a deep link has seconds in the query parameters,then the datetime-local type will show seconds in the datepicker.

File: client/app/services/query.js [hunk 2]
```diff
@@ -43,6 +43,40 @@ class QueryResultError {
...
+      this.$$value = this.$$value || parseInt(this.value, 10);
+      return this.$$value;
+    }
+
+    return this.value;
+  }
+
+  set ngModel(value) {
+    if (this.type === 'date') {
+      this.value = moment(value).format('YYYY-MM-DD');
```
Comment (rockwotj) [inline client/app/services/query.js]: I dislike having this hardcoded, why not use the environment variable?

File: client/app/services/query.js (hunk 1 above)
Comment (arikfr) [inline client/app/services/query.js]: Wasn't aware of this - fixed.

File: client/app/services/query.js (hunk 2 above)
Comment (arikfr) [inline client/app/services/query.js]: For now this maintains backward compatibility with v0.12.0 and fixes a regression. We can revisit this and add option to set the format in a future pull request.

But I don't think it should be in an env variable, but rather as configuration for the parameter, as users might use different databases on the same Redash deployment.

File: client/app/services/query.js [hunk 3]
```diff
@@ -43,6 +43,43 @@ class QueryResultError {
...
+
+    return this.value;
+  }
+
+  set ngModel(value) {
+    if (value && this.type === 'date') {
+      this.value = moment(value).format('YYYY-MM-DD');
+      this.$$value = moment(this.value).toDate();
+    } else if (value && this.type === 'datetime-local') {
+      this.value = moment(value).format('YYYY-MM-DDTHH:mm');
```
Comment (rockwotj) [inline client/app/services/query.js]: BigQuery sadly doesn't support this format (it doesn't like the `T`) so we won't be able to use this (we've been using the fix in #1553) until it's configurable.

Comment (arikfr) [inline client/app/services/query.js]: What format BQ supports?

Comment (rockwotj) [inline client/app/services/query.js]: `YYYY-MM-DD HH:mm:ss`

Comment (arikfr) [inline client/app/services/query.js]: I changed it to be the current format :)

**Comment by rockwotj** [comment]
Whoop! Thanks Arik :)

**Comment by rockwotj** [comment]
Thanks Arik :)

**Comment by arikfr** [comment]
Sure thing. Thank you for the feedback and help here!

**Pull request merged**